module5/lessons/Week1/RefactoringPatterns.md

```diff
@@ -0,0 +1,415 @@
+---
+layout: page
+title: SOLID Design Principles and Refactoring Patterns
+---
+
+## Learning Goals
+* Be able to name the SOLID design principles
+* Explore a few refactoring patterns
+* Understand that there are _many_ more refactoring patterns
+
+# SOLID
+
+When writing code, there are many things to consider, including functionality, usability, and security.  Another important consideration is the **design** of our code.  There are _many_ strategies that have been developed to help us design understandable, flexible and maintainable code.  Today, we are going to dive into 5 of the most common: the **SOLID** principles.
+
+**S**: Single-Responsibility  
+**O**: Open-Closed  
+**L**: Liskov Substitution  
+**I**: Interface Segregation  
+**D**: Dependency Inversion  
+
+<section class='instructor-notes' markdown='1'>
+
+Here is a good resource on the SOLID principles using c# examples: https://www.c-sharpcorner.com/UploadFile/damubetha/solid-principles-in-C-Sharp/
+
+</section>
+
+<section class='call-to-action' markdown='1'>
+
+**Small Group Round Table**
```

Comment:
The Solid principles are tricky, but I like that we're touching on them a bit and in general thinking about maintainable code. I like that you clarified that this is definitely not a topic they should be an expert on.